File: GameHub/src/app/models/api-models/Game.ts
export interface Game{
    id:number,
    name:string,
    price:number,
    imageUrl:string,
    gameDescription:string,
    date: Date,
    rating:number,
    status:string,
    genreId:number,
    platformId:number
    genre: {
        id:number,
        name:string
    },
    platform: {
        id:number,
        name:string
    }
}

File: GameHubPortalApi/GameHubPortalApi/DataModels/Game.cs
using System.ComponentModel.DataAnnotations;

namespace GameHubPortalApi.DataModels
{
    public class Game
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public double Price { get; set; }

        public string ImageUrl { get; set; }

        public DateTime Date { get; set; }
        public string GameDescription { get; set; }

        [Range(1,5)]
        public int Rating { get; set; }
        public string? Status { get; set; }

        public int GenreId { get; set; }

        public List<Genre> Genres { get; set; } = new List<Genre>();

        public int PlatformId { get; set; }

        public Platform Platform { get; set; }

    }
}
